src/unit.c

```diff
@@ -50,7 +50,7 @@ unit_new(const gchar *name,
     symbol_names = g_strsplit(symbols, ",", 0);
     for (i = 0; symbol_names[i]; i++)
         unit->priv->symbols = g_list_append(unit->priv->symbols, g_strdup(symbol_names[i]));
-    g_free(symbol_names);
+    g_strfreev(symbol_names);
```

Comment:
A possibly better fix would be not to call `g_strdup()` in the `g_list_append()` call, so it'd avoid unnecessary memory duplication. Then keep the `g_free()` to only free the array itself and not the items.

Reply:
:+1: